randovania/generator/base_patches_factory.py

```diff
@@ -6,259 +6,123 @@
     pass
 
-
-def add_elevator_connections_to_patches(configuration: EchoesConfiguration,
-                                        rng: Random,
-                                        patches: GamePatches) -> GamePatches:
-    """
-    :param configuration:
-    :param rng:
-    :param patches:
-    :return:
-    """
-    elevator_connection = copy.copy(patches.elevator_connection)
-    elevators = configuration.elevators
-
-    if not elevators.is_vanilla:
-        if rng is None:
-            raise MissingRng("Elevator")
-
-        world_list = default_database.game_description_for(configuration.game).world_list
-        elevator_db = elevator_distributor.create_elevator_database(
-            world_list, elevators.editable_teleporters)
-
-        if elevators.mode in {TeleporterShuffleMode.TWO_WAY_RANDOMIZED, TeleporterShuffleMode.TWO_WAY_UNCHECKED}:
-            connections = elevator_distributor.two_way_elevator_connections(
-                rng=rng,
-                elevator_database=elevator_db,
-                between_areas=elevators.mode == TeleporterShuffleMode.TWO_WAY_RANDOMIZED
-            )
+class BasePatchesFactory:
+    def create_base_patches(self,
+                            configuration: BaseConfiguration,
+                            rng: Random,
+                            game: GameDescription,
+                            is_multiworld: bool,
+                            player_index: int,
+                            ) -> GamePatches:
+        """
+        """
+        patches = dataclasses.replace(game.create_game_patches(),
+                                    player_index=player_index)
+
+        # Elevators
+        if hasattr(configuration, "elevators"):
+            patches = self.add_elevator_connections_to_patches(configuration, rng, patches)
+
+        # Configurable Nodes
+        patches = patches.assign_node_configuration(
+            self.configurable_node_assignment(configuration, game, rng)
+        )
+
+        # Starting Location
+        patches = patches.assign_starting_location(
+            self.starting_location_for_configuration(configuration, game, rng))
+
+        # Hints
+        if rng is not None:
+            patches = self.add_default_hints_to_patches(rng, patches, game.world_list,
+                                                        num_joke=self.num_joke_hints, is_multiworld=is_multiworld)
+
+        return patches
+    
+    def add_elevator_connections_to_patches(self,
+                                            configuration: BaseConfiguration,
+                                            rng: Random,
+                                            patches: GamePatches) -> GamePatches:
+        """
+        :param configuration:
+        :param rng:
+        :param patches:
+        :return:
+        """
+        elevator_connection = copy.copy(patches.elevator_connection)
+        elevators = configuration.elevators
+
+        if not elevators.is_vanilla:
```

Comment:
The implementation in this function is a Prime/Echoes/Corruption specific one, since the `elevators` field is not present in BaseConfiguration.

Reply:
:+1: